### PR TITLE
inject: fix non-oneshot music triggers

### LINKED ISSUE #939: OG bug: several music triggers are not one shot
In relation to #905 and #755, the following music triggers should be converted to one shot via injections.

| Level | Triggers |
|-|-|
| Caves | Room 34/track 9 |
| Tomb of Qualopec | Room 25/track 17 |
| St. Francis' Folly | Room 18/track 7 |
| Obelisk of Khamoon | Room 12/track 3<br/>Room 32/track 4 |
| Sanctuary of the Scion | Room 0/track 10 |
| Natla's Mines | Room 86/track 3 |
| Atlantis | Room 59/track 8 |
| The Great Pyramid | Room 36/track 8 |
| Return to Egypt | Room 0/track 19<br/>Room 15/track 14<br/>Room 19/track 15<br/>Room 22/track 16<br/>Room 61/track 6<br/>Room 93/track 11  |
| Temple of the Cat | Room 14/track 12<br/>Room 98/track 7<br/>Room 100/track 20 |
| Atlantean Stronghold | Room 4/track 20<br/>Room 13/track 19<br/>Room 17/track 11<br/>Room 20/track 15<br/>Room 25/track 12 |
| The Hive | Room 8/track 9<br/>Room 18/track 6<br/>Room 30/track 12<br/>Room 31/track 18<br/>Room 32/track 3<br/>Room 35/track 20 |

### PR DESCRIPTION
**Pull request by lahm86**
Resolves #939.

#### Checklist

- [x] I have read the [coding conventions](https://github.com/rr-/Tomb1Main/blob/master/CONTRIBUTING.md#coding-conventions)
- [x] I have added a changelog entry about what my pull request accomplishes, or it is an internal change

#### Description

Sets the one shot flag on those triggers identified in #755. We do this with a guard in case the injections are used on levels without music triggers where we expect them to be - so only if a trigger on a sector has a `TO_CD` command, will the trigger itself become one shot.

The ones affected are mostly triggers with only music commands, although some also have a `FLIP_OFF` command, such as in Folly. I've tested these and the flipmaps still behave as they should.

### CODE REVIEW

**Review by rr- — APPROVED** [review]

**Review by walkawayy — APPROVED** [review]

File: src/game/inject.c [hunk 1]
```diff
@@ -1076,6 +1081,63 @@ static void Inject_TriggerParameterChange(
...
+        case FT_ROOF:
+        case FT_TILT:
+            fd_index++;
+            break;
+
+        case FT_LAVA:
+            break;
+
+        case FT_TRIGGER: {
+            uint16_t trig_type = (data & 0x3F00) >> 8;
```
Comment (rr-) [inline src/game/inject.c]: Can we use `TRIG_BITS` here?

Comment (lahm86) [reply src/game/inject.c]: `TRIG_BITS` is `(T & 0x3C00) >> 10` - should we make a new `define` for this instead? e.g. `TRIG_TYPE`

Comment (walkawayy) [reply src/game/inject.c]: So `TRIG_BITS` triggers the floor trigger and `TRIG_TYPE` would check the floor trigger type?

Comment (lahm86) [reply src/game/inject.c]: The first `int16_t` is the data type packed with the trigger type (other engines pack things like triangulation types for TR3 triangles and suchlike here too). So e.g.:

`data = g_FloorData[100];`
`0000000011111111`
`data & 0x00FF` =>  entry type - floor, tilt, trigger etc

`0011111100000000`
`(data & 0x3F00) >> 8` => trigger type - pad, heavy, switch etc

`1000000000000000`
`data & 0x8000` => end bit (no more entries for this tile)

Then we move to the next `int16_t` in the array for the trigger's commands. Those are packed differently. `TRIG_BITS` pulls out the command type. The other packed value is the parameter. So e.g.:

`command = g_FloorData[101];`
`0011110000000000`
`(command & 0x3C00) >> 10` => command type - object, music, flipmap etc

`0000001111111111`
`command & 0x03FF` => command parameter - item 7 or music track 7 etc

`1000000000000000`
`command & 0x8000` => end bit (no more commands for this trigger)

Bit 14 is unused in TR1 in both cases.

Comment (rr-) [reply src/game/inject.c]: Yeah let's introduce a new const for this; later on we can refactor it but definitely should avoid using magic constants across code.

Comment (lahm86) [reply src/game/inject.c]: Const added, let me know if we're OK to merge.

**Review by rr- — APPROVED** [review]

**Review by walkawayy — APPROVED** [review]

**Pull request merged**